src/presence_analyzer/tests.py

```diff
@@ -355,6 +355,23 @@ def test_group_start_end_by_weekday(self):
 
         self.assertEqual(result, utils.group_start_end_by_weekday(fake_data))
 
+    def test_cache_decorator(self):
+        """
+        Test @cache result
+        """
+        def gen_num():
+            for _ in range(1, 3):
+                yield _
+
+        num = gen_num()
+
+        @utils.cache(300)
+        def test_func(x):
+            return x
+
+        self.assertEqual(test_func(num.next()), 1)
+        self.assertEqual(test_func(num.next()), 2)
```

Comment:
Unfortunately this doesn't really test anything, we might not understood each other, how do we check cache, if we ask every time with different args? Should be asked multiple times with same and different args to check if it does what it should. We can do even better by checking internal call count using mock.

I've written some in-rush code, look:
```
        moc = mock.MagicMock()
        _func = lambda _: moc()
        cachelimit = 0.001

        wrapped = utils.cache(cachelimit)(_func)

        # check first two unique arg calls
        wrapped(1)
        wrapped(2)
        self.assertEqual(moc.call_count, 2)

        # check if third non-unique arg call won't call the mock
        wrapped(1)
        self.assertEqual(moc.call_count, 2)

        # check if after cache time limit it calls the func again
        import time
        time.sleep(cachelimit * 2)
        wrapped(1)
        self.assertEqual(moc.call_count, 3)
```
See, i've tested 3 things:
- if it caches whatever is in func
- if it returns cached in time
- if it doesn't return cached after time limit

Rewrite your code to test those, use some of mine, you can make it better by not using time lib (and relay on thread performance) but mocking `datetime.now` at appropriate point.